go/fn/resourcelist.go

```diff
@@ -57,6 +57,21 @@ type ResourceList struct {
 	Results Results `yaml:"results,omitempty" json:"results,omitempty"`
 }
 
+// CheckResourceDuplication checks the GVKNN of resourceList.items to make sure they are unique. It returns errors if
+// found more than one resource having the same GVKNN.
+func CheckResourceDuplication(rl *ResourceList) error {
+	idMap := map[yaml.ResourceIdentifier]bool{}
```

Comment:
nit: It seems `bool` is not useful, since it's always true. We can change it to `idMap := map[yaml.ResourceIdentifier]struct{}{}`

Reply:
Ah! Thank you so much!  Didn't know that
https://itnext.io/set-in-go-map-bool-and-map-struct-performance-comparison-5315b4b107b